knora/dsplib/models/user.py

```diff
@@ -475,18 +476,17 @@ def fromJsonObj(cls, con: Connection, json_obj: Any):
         in_groups: set[str] = set()
         if json_obj.get('permissions') is not None and json_obj['permissions'].get('groupsPerProject') is not None:
             sysadmin = False
-            project_groups = json_obj['permissions']['groupsPerProject']
-            for project in project_groups:
-                if project == Project.SYSTEM_PROJECT:
-                    if Group.PROJECT_SYSTEMADMIN_GROUP in project_groups[project]:
+            for project_iri, group_memberships in json_obj['permissions']['groupsPerProject'].items():
+                if project_iri == Project.SYSTEM_PROJECT:
+                    if Group.PROJECT_SYSTEMADMIN_GROUP in group_memberships:
                         sysadmin = True
                 else:
-                    for group in project_groups[project]:
+                    for group in group_memberships:
                         if group == Group.PROJECT_MEMBER_GROUP:
-                            if in_projects.get(project) is None:
-                                in_projects[project] = False
+                            if in_projects.get(project_iri) is None:
+                                in_projects[project_iri] = False
                         elif group == Group.PROJECT_ADMIN_GROUP:
-                            in_projects[project] = True
+                            in_projects[project_iri] = True
```

Comment:
This is just a refactoring, without semantic modification.

Reply:
nice!